include/envoy/config/grpc_mux.h

```diff
@@ -100,6 +100,9 @@ class GrpcMux {
                                    const std::set<std::string>& resources,
                                    SubscriptionCallbacks& callbacks,
                                    OpaqueResourceDecoder& resource_decoder) PURE;
+
+protected:
+  absl::flat_hash_map<std::string, std::string> type_url_mapping_;
```

Comment:
Should we put this on interface? I think that this information should not have here.

Reply:
Sure, will move it to impl file.

Reply:
Might make sense to move to an implementation file local `CONSTRUCT_ON_FIRST_USE` singleton, since it's only accesses on the main thread and the type mapping doesn't change.

Reply:
Yes, that makes sense.